app/lib/frontend/handlers/listing.dart

```diff
@@ -58,31 +60,31 @@ Future<shelf.Response> webPackagesHandlerHtml(shelf.Request request) async {
 /// - /packages - package listing
 /// - /dart/packages
 /// - /flutter/packages
-/// - /server/packages
-/// - /web/packages
 Future<shelf.Response> _packagesHandlerHtmlCore(shelf.Request request,
-    {String platform, String sdk}) async {
+    {String sdk}) async {
   // TODO: use search memcache for all results here or remove search memcache
   final searchQuery = parseFrontendSearchQuery(
     request.requestedUri.queryParameters,
-    platform: platform,
     sdk: sdk,
     tagsPredicate: TagsPredicate.regularSearch(),
   );
+  final searchPreferences = SearchPreference.fromSearchQuery(searchQuery);
   final sw = Stopwatch()..start();
   final searchResult = await searchAdapter.search(searchQuery);
   final int totalCount = searchResult.totalCount;
 
   final links =
       PageLinks(searchQuery.offset, totalCount, searchQuery: searchQuery);
-  final result = htmlResponse(renderPkgIndexPage(
-    searchResult.packages,
-    links,
-    platform: platform,
-    sdk: sdk,
-    searchQuery: searchQuery,
-    totalCount: totalCount,
-  ));
+  final result = htmlResponse(
+    renderPkgIndexPage(
+      searchResult.packages,
+      links,
+      sdk: sdk,
+      searchQuery: searchQuery,
+      totalCount: totalCount,
+    ),
+    headers: createSearchPreferenceCookie(searchPreferences),
```

Comment:
Can this conflict with other cookies? Just thinking...

Reply:
We only set a single cookie at any response, and it does not causes conflicts with already set cookies.

Reply:
I see no reason to fear conflicts :)